### DISCUSSION TITLE
Deprecated: Optional parameter | implicitly treated as a required parameter

### DISCUSSION THREAD

**Discussion by wissedenis** [timschofield/webERP#317] [Problems / Bugs]
hello everythin, 
has anyone faced this issue 
php8.1
Deprecated: Optional parameter $Disc declared before required parameter $Weight is implicitly treated as a required parameter in /includes/DefineCartClass.php on line 80
Deprecated: Optional parameter $UpdateDB declared before required parameter $identifier is implicitly treated as a required parameter in /DefineCartClass.php on line 183

even when you change the order the error stays! It keeps on changing,
this is really annoying

**Comment by timschofield** [comment]
Hi, I assume you are using 4.15.2? This version isn't php8 compatible and your error is php8 warning. This problem has been fixed in the master branch.
You can avoid seeing this message by changing your error_reporting() setting in config.php, or by using php7

Thanks
Tim

**Reply by wissedenis** [reply]
Hello, thanks for the reply.
The master branch has it's own problem, I tried both fresh installation and upgrade but stack on Z_UpgradeDatabase.php even when you try to go around by creating the column manually "tag" into "gltags" it just stop from there 
https://github.com/timschofield/webERP/issues/311

**Reply by wissedenis** [reply]
> Hi, I assume you are using 4.15.2? This version isn't php8 compatible and your error is php8 warning. This problem has been fixed in the master branch. You can avoid seeing this message by changing your error_reporting() setting in config.php, or by using php7
> 
> Thanks Tim

I also got DefinePOClass.php from the master  branch and replaced it but the error persists

**Comment by timschofield** [comment]
The error you were citing was in includes/DefineCartClass.php not in
DefinePOClass.php

On Fri, 20 Dec 2024 at 05:51, wissedenis ***@***.***> wrote:

> Hi, I assume you are using 4.15.2? This version isn't php8 compatible and
> your error is php8 warning. This problem has been fixed in the master
> branch. You can avoid seeing this message by changing your
> error_reporting() setting in config.php, or by using php7
>
> Thanks Tim
>
> I also got DefinePOClass.php from the master branch and replaced it but
> the error persists
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/317#discussioncomment-11624844>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LCTKZPNMJN624CCRKL2GOV7XAVCNFSM6AAAAABT2U5BGKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRSGQ4DINA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/
